### PR TITLE
Fix endpoint name in docs

### LINKED ISSUE #4022: deliveryservice_server_capabilities Doc requires to be fixed with endpoint info
## I'm submitting a ...
- [X] bug report
- [ ] new feature / enhancement request
- [ ] improvement request (usability, performance, tech debt, etc.)
- [ ] other 

## Traffic Control components affected ...
- [ ] CDN in a Box
- [X] Documentation
- [ ] Grove
- [ ] Traffic Control Client
- [ ] Traffic Monitor
- [ ] Traffic Ops
- [ ] Traffic Ops ORT
- [ ] Traffic Portal
- [ ] Traffic Router
- [ ] Traffic Stats
- [ ] Traffic Vault
- [ ] unknown

## Current behavior:
[The documentation](https://traffic-control-cdn.readthedocs.io/en/latest/api/deliveryservice_server_capabilities.html) refers to `deliveryservices_required_capabilities` as deliveryservice_server_capabilities

## Expected / new behavior:

deliveryservice_server_capabilities should be `deliveryservices_required_capabilities`

## Anything else:

Response Type for Delete is  undefined. Should it be object?

### PR DESCRIPTION
**Pull request by ChrisHines**
## What does this PR (Pull Request) do?
- [x] This PR fixes #4022

## Which Traffic Control components are affected by this PR?
- Documentation

## What is the best way to verify this PR?
Build the docs and check that the API docs for `deliveryservice_server_capabilities` are renamed to `deliveryservices_required_capabilities`.

## If this is a bug fix, what versions of Traffic Control are affected?
- master (426881af)

## The following criteria are ALL met by this PR
- [x] This PR includes tests OR I have explained why tests are unnecessary
- [x] This PR includes documentation OR I have explained why documentation is unnecessary
- [x] This PR includes an update to CHANGELOG.md OR such an update is not necessary
- [x] This PR includes any and all required license headers
- [x] This PR ensures that database migration sequence is correct OR this PR does not include a database migration
- [x] This PR **DOES NOT FIX A SERIOUS SECURITY VULNERABILITY** (see [the Apache Software Foundation's security guidelines](https://www.apache.org/security/) for details)

## Additional Information
Related endpoint implementation in #4026.

### CODE REVIEW

File: docs/source/api/deliveryservices_required_capabilities.rst [hunk 1]
```diff
@@ -13,11 +13,11 @@
 .. limitations under the License.
 ..
 
-.. _to-api-deliveryservice-server-capabilities:
+.. _to-api-deliveryservice-required-capabilities:
```
Comment (mhoppa) [inline docs/source/api/deliveryservices_required_capabilities.rst]: should this be deliveryservices-required-capabilites? ( missing s on end of deliveryservice)

Comment (ChrisHines) [reply docs/source/api/deliveryservices_required_capabilities.rst]: I guess so. What does this line get used for in the docs anyway?

Comment (mhoppa) [reply docs/source/api/deliveryservices_required_capabilities.rst]: I believe it is used to reference from other pages

Comment (ocket8888) [reply docs/source/api/deliveryservices_required_capabilities.rst]: yes, that's a reference label. You can, in practice put them anywhere to act as labels ("anchors" in HTML) but truly "label-able" things let you do
```rst
.. _section:

*************
SECTION TITLE
*************
Some paragraph text

.. _some-thing:

lorem ipsum dolor etc.

.. _figure:
.. figure:: jay-peg.png
    :caption: Figure caption

.. _table:
.. table:: My table

    A | B
    -+-
    1| 0

:ref:`section` (renders as "SECTION TITLE")
:ref:`figure` (renders as "Figure caption")
:ref:`table` (renders as "My table")
:ref:`some-thing` (this is an error - no title)
:ref:`Some thing <some-thing>` (Renders as a link to some-thing with the text "Some thing")
```

**Comment by ChrisHines** [comment]
> I think maybe you also wanna change the success alert in the POST handler. It currently says "deliveryservice server_capability was updated."

If I'm reading ["How to Read this Documentaion"](https://traffic-control-cdn.readthedocs.io/en/latest/api/index.html#response-structure) right the "alerts" should be removed from the response structure example.

**Comment by ocket8888** [comment]
Nah, that's just about the field lists that describe the structure:

> _"The field definitions listed in the Response Structure subsection of an endpoint method are the elements of this object"_

Examples should be complete requests and responses to that request. You just don't need to lay out "this endpoint returns an alerts array with one alert that has these properties..." because that's repetitive, and unnecessary. Honestly, it doesn't even really matter if the alerts shown aren't accurate, but the specific wording in this case could make it seem like that response is for a different endpoint.

**Comment by ChrisHines** [comment]
Correct me if this is wrong. As I read the code for #4026, it uses [`api.CreateHandler`](https://github.com/apache/trafficcontrol/blob/master/traffic_ops/traffic_ops_golang/api/shared_handlers.go#L310) to implement the POST request, which will format the alert message as `"deliveryservice.RequiredCapability was created."`

The alert in the DELETE request needed a similar fix, plus updating of new-line placement to be consistent with the other examples.

**Pull request merged**